src/components/Menus/WalletSelectMenu/WalletSelectMenuContent.tsx

```diff
@@ -0,0 +1,180 @@
+import {
+  MenuHeaderAppBar,
+  MenuHeaderAppWrapper,
+  MenuList,
+  MenuPaper,
+} from '@/components/Menu/Menu.style';
+import { MenuKeysEnum } from '@/const/menuKeys';
+import {
+  type Breakpoint,
+  darken,
+  type SxProps,
+  type Theme,
+  Typography,
+} from '@mui/material';
+import { useMenuStore } from '@/stores/menu';
+import type { ReactNode } from 'react';
+import {
+  EcosystemSelectMenu,
+  useWalletSelectContent,
+} from '@/components/Menus';
+import { MenuItem } from '@/components/Menu';
+import { getContrastAlphaColor } from '@/utils/colors';
+import { WalletSelectMoreSubMenu } from '@/components/Menus/WalletSelectMoreSubMenu';
+import { useTheme } from '@mui/material/styles';
+import { useTranslation } from 'react-i18next';
+
+const NUMBER_OF_WALLETS_DISPLAYED = 9;
+
+interface WalletSelectMenuContentProps {
+  openWalletSelectMenu: boolean;
+  showAllButton?: boolean;
+}
+
+function WalletSelectMenuContent({
+  openWalletSelectMenu,
+  showAllButton = true,
+}: WalletSelectMenuContentProps) {
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const isDarkMode = theme.palette.mode === 'dark';
+  const walletSelectMenuItems = useWalletSelectContent();
+  const filteredWalletSelectMenuItems = walletSelectMenuItems.slice(
+    0,
+    NUMBER_OF_WALLETS_DISPLAYED,
+  );
+  const { openSubMenu, setSubMenuState } = useMenuStore((state) => state);
+
+  const menuItemStyles: SxProps<Theme> = {
+    margin: 0,
```

Comment:
should be its own component

Reply:
I agree but not easy at all because of how the MenuItem overrided component has been done, I did not had anything but moving the logic to WalletSelectMenuContent, original is there https://github.com/jumperexchange/jumper-exchange/blob/develop/src/components/Menus/WalletSelectMenu/WalletSelectMenu.tsx#L39